src/raw/windows.rs

```diff
@@ -345,16 +358,27 @@ impl Socket {
     }
 
     ///Accepts incoming connection.
-    pub fn accept(&self) -> io::Result<(Socket, net::SocketAddr)> {
+    pub fn accept(&self, flags: c_int) -> io::Result<(Socket, net::SocketAddr)> {
+        if (flags & !(winapi::SOCK_NONBLOCK | winapi::SOCK_CLOEXEC)) != 0 {
+            // `accept4()` also returns `EINVAL` if it receives an unknown flag
+            return Err(io::Error::from_raw_os_error(winapi::WSAEINVAL));
+        }
+
         unsafe {
             let mut storage: winapi::SOCKADDR_STORAGE_LH = mem::zeroed();
             let mut len = mem::size_of_val(&storage) as c_int;
 
             match winapi::accept(self.inner, &mut storage as *mut _ as *mut _, &mut len) {
                 winapi::INVALID_SOCKET => Err(io::Error::last_os_error()),
                 sock @ _ => {
+                    let sock = Socket { inner: sock };
                     let addr = sockaddr_to_addr(&storage, len)?;
-                    Ok((Socket { inner: sock, }, addr))
+
+                    // Emulate the two most common (and useful) `accept4` flags using `ioctl`/`fcntl`

```

Comment:
Comment on emulation isn't needed as i believe it is already clear from existence of flags.
But please add description of available flags to public documentation